src/Symfony/Component/Yaml/Inline.php

```diff
@@ -463,8 +463,8 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = ar
                 break;
             }
 
-            if (!isset($mapping[$i + 1]) || ' ' !== $mapping[$i + 1]) {
-                @trigger_error('Omitting the space after the colon that follows a mapping key definition is deprecated since version 3.2 and will throw a ParseException in 4.0.', E_USER_DEPRECATED);
+            if (!isset($mapping[$i + 1]) || !in_array($mapping[$i + 1], array(' ', '[', ']', '{', '}'), true)) {
+                @trigger_error('Using a colon that is not followed by an indication character (i.e. " ", ",", "[", "]", "{", "}" is deprecated since version 3.2 and will throw a ParseException in 4.0.', E_USER_DEPRECATED);
```

Comment:
That would be misleading as the current outcome is depending on the actual YAML input. You may get an exception, but the colon might also be treated as part of the mapping key.